src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ShowFeedbackAction.java

```diff
@@ -0,0 +1,194 @@
+
+  @NotNull
+  private final MainViewModelProvider mainViewModelProvider;
+
+  @NotNull
+  private final Interfaces.AuthenticationProvider authenticationProvider;
+
+  @Nullable
+  private SubmissionResult submissionResult = null;
+
+  @NotNull
+  private final Notifier notifier;
+
+  /**
+   * Default constructor.
+   */
+  public ShowFeedbackAction() {
+    this(
+        PluginSettings.getInstance(),
+        project -> Optional.ofNullable(PluginSettings.getInstance().getCourseProject(project))
+            .map(CourseProject::getAuthentication).orElse(null),
+        new DefaultNotifier()
+    );
+  }
+
+  /**
+   * Constructor.
+   */
+  public ShowFeedbackAction(
+      @NotNull MainViewModelProvider mainViewModelProvider,
+      @NotNull Interfaces.AuthenticationProvider authenticationProvider,
+      @NotNull Notifier notifier) {
+    this.mainViewModelProvider = mainViewModelProvider;
+    this.authenticationProvider = authenticationProvider;
+    this.notifier = notifier;
+
+  }
+
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e) {
+    var project = e.getProject();
+    if (project == null) {
+      return;
+    }
+
+    var mainViewModel = mainViewModelProvider.getMainViewModel(project);
+    var exercisesViewModel = mainViewModel.exercisesViewModel.get();
+    var courseViewModel = mainViewModel.courseViewModel.get();
+    var authentication = authenticationProvider.getAuthentication(project);
+
+    if (courseViewModel == null || exercisesViewModel == null || authentication == null) {
+      return;
+    }
+
+    var selection = (ExercisesTreeViewModel.ExerciseTreeSelection) exercisesViewModel.findSelected();
+    var selectedExercise = selection.getExercise();
+    var selectedSubmission = selection.getSubmissionResult();
+    if ((selectedExercise == null || selectedSubmission == null) && submissionResult == null) {
+      return;
+    }
+    if (submissionResult == null) {
+      submissionResult = selectedSubmission.getModel();
+    }
+    var course = courseViewModel.getModel();
+    var exerciseDataSource = course.getExerciseDataSource();
+
+    var progress = mainViewModel.progressViewModel.start(1, "Loading Feedback...", true);
+    try {
+
+      var feedbackString = exerciseDataSource.getSubmissionFeedback(submissionResult.getId(), authentication);
+      var document = Jsoup.parseBodyFragment(feedbackString);
+
+      var textColor = JBColor.black;
+      var textColorString =
+          String.format("#%02x%02x%02x", textColor.getRed(), textColor.getGreen(), textColor.getBlue());
+      var fontName = JBFont.regular().getFontName();
+
+      switch (course.getName()) {
+        case "O1":
+          var style = document.select("style").first();
+          if (style != null) {
+            style.remove();
+          }
+          document.head().append(
+              "<style>body {  color: " + textColorString
```

Comment:
Idea: Could course config files point to LESS(?) files that describe the feedback style and may refer to (a predefined set of) variables such as `fontName` and `textColorString`. Then the plugin would download that file (if there's one), compile it to CSS with appropriate variable values and put into a `<style>` element of each feedback. That way, I think, this whole feature could be generic and not course-specific.